concent_api/verifier/tests/test_integration_verification.py

```diff
@@ -240,5 +272,200 @@ def test_blender_verification_order_should_call_verification_result_with_result_
             VerificationResult.ERROR,
             'error',
-            'verifier.blender_verification_order.running_blender',
+            ErrorCode.VERIFIIER_RUNNING_BLENDER_FAILED.name,
+        )
+
+    def test_blender_verification_order_should_call_verification_result_with_result_error_if_opening_first_file_raise_memory_error(self):
+        with mock.patch('verifier.tasks.clean_directory', autospec=True) as mock_clean_directory,\
+            mock.patch('verifier.tasks.send_request_to_storage_cluster', autospec=True) as mock_send_request_to_storage_cluster,\
+            mock.patch('verifier.tasks.store_file_from_response_in_chunks', autospec=True) as mock_store_file_from_response_in_chunks,\
+            mock.patch('verifier.tasks.unpack_archive', autospec=True) as mock_unpack_archive,\
+            mock.patch('verifier.tasks.verification_result.delay', autospec=True) as mock_verification_result,\
+            mock.patch('verifier.tasks.run_blender', mock_run_blender), \
+            mock.patch('builtins.open', autospec=True, side_effect=MemoryError('error')), \
+            mock.patch('verifier.tasks.get_files_list_from_archive', return_value=['file_name']) as mock_get_files_list_from_archive, \
+            mock.patch('verifier.tasks.delete_file') as mock_delete_file:  # noqa: E125
+            blender_verification_order(
+                subtask_id=self.compute_task_def['subtask_id'],
+                source_package_path=self.source_package_path,
+                source_size=self.report_computed_task.task_to_compute.size,
+                source_package_hash=self.report_computed_task.task_to_compute.package_hash,
+                result_package_path=self.result_package_path,
+                result_size=self.report_computed_task.size,  # pylint: disable=no-member
+                result_package_hash=self.report_computed_task.package_hash,  # pylint: disable=no-member  # pylint: disable=no-member
+                output_format=BlenderSubtaskDefinition.OutputFormat(
+                    self.compute_task_def['extra_data']['output_format']
+                ).name,
+                scene_file=self.compute_task_def['extra_data']['scene_file'],
+            )
+
+        self.assertEqual(mock_clean_directory.call_count, 1)
+        self.assertEqual(mock_send_request_to_storage_cluster.call_count, 2)
+        self.assertEqual(mock_store_file_from_response_in_chunks.call_count, 2)
+        self.assertEqual(mock_unpack_archive.call_count, 2)
+        self.assertEqual(mock_get_files_list_from_archive.call_count, 1)
+        self.assertEqual(mock_delete_file.call_count, 2)
+        mock_verification_result.assert_called_once_with(
+            self.compute_task_def['subtask_id'],
+            VerificationResult.ERROR,
+            'error',
+            ErrorCode.VERIFIIER_LOADING_FILES_INTO_MEMORY_FAILED.name,
+        )
+
+    def test_blender_verification_order_should_continue_verification_if_opening_first_file_raise_os_error(self):
```

Comment:
BTW: I am not sure if this is proper behaviour - if reading the file fails and it cannot be uploaded, why should verification continue?

Reply:
I agree. The verification should crash if the file cannot be opened.

The only case where we want to ignore problems with a file is when we try to upload our rendering result to the storage server. It's only for auditing purposes so we can ignore it and continue verification. Any other file is critical for us and a problem should result either in a crash (if we're sure this could have happened only due to a bug) or a verification error reported to the client (if it's client's fault, e.g. invalid file type or an empty file).

BTW, I see a typo in the code above: `VERIFIIER` -> `VERIFIER`.

Reply:
It is working ok, just the test name was misleading, `first_file` meant blender output file in this context, I have changed the test name.